qiskit/extensions/quantum_initializer/ucrx.py

```diff
@@ -35,29 +32,31 @@ class UCRXGate(UCPauliRotGate):
     The decomposition is based on
     'Synthesis of Quantum Logic Circuits' by V. Shende et al.
     (https://arxiv.org/pdf/quant-ph/0406176.pdf)
-
-    Input:
-    angle_list = list of (real) rotation angles [a_0,...,a_{2^k-1}]
     """
 
     def __init__(self, angle_list):
         super().__init__(angle_list, "X")
 
 
-def ucrx(self, angle_list, q_controls, q_target):
-    """Attach a uniformly controlled (also called multiplexed) Rx rotation gate to a circuit.
+def ucrx(
+    self,
+    angle_list: List[float],
+    q_controls: Union[QuantumRegister, List[Qubit]],
+    q_target: Union[QuantumRegister, Qubit],
```

Comment:
The qubits here could also be integers right? It would be nice to also use the `QubitSpecifier` type that the gates in `quantumcircuit.py` are using. Also we're moving to the built-in typing instead of using the explicit typing module, so the type hint would be something like
```python
# at top of file
from __future__ import annotations
from collections.abc import Sequence

...
q_controls: Sequence[QubitSpecifier],
q_target: QubitSpecifier
```
🙂

Reply:
Thank you for the review & suggestions! I made the changes